lib/resty/upstream/healthcheck.lua

```diff
@@ -230,16 +230,26 @@ local function check_peer(ctx, id, peer, is_backup)
         ok, err = sock:connect(name)
     end
     if not ok then
-        if not peer.down then
-            errlog("failed to connect to ", name, ": ", err)
+        return peer_error(ctx, is_backup, id, peer, "failed to connect to ", name, ": ", err)
+    end
+
+    if ctx.type == "https" then
+        session, err = sock:sslhandshake(ctx.session, name, ctx.ssl_verify)
```

Comment:
Better use the `lua-releng` script of `nginx-devel-utils` to check your Lua code. I'm getting the following errors by running this tool against your branch:

```
$ lua-releng
lib/resty/upstream/healthcheck.lua: 0.03 (0.03)
Checking use of Lua global variables in file lib/resty/upstream/healthcheck.lua...
	op no.	line	instruction	args	; code
	53	[237]	SETGLOBAL	11 -14	; session
	54	[238]	GETGLOBAL	11 -14	; session
	73	[244]	GETGLOBAL	11 -14	; session
Checking line length exceeding 80...
```

So apparently the `session` is a misused Lua global variable.